web/components/ConferenceUpdatesForm/ConferenceUpdatesForm.tsx

```diff
@@ -1,55 +1,69 @@
+import { useRouter } from 'next/router';
+import { PortableTextComponentProps } from '@portabletext/react';
 import GridWrapper from '../GridWrapper';
-import Heading from '../Heading';
 import PlaceholderImage from '../PlaceholderImage';
 import styles from './ConferenceUpdatesForm.module.css';
-import { PortableTextComponentProps } from '@portabletext/react';
 
 type ConferenceUpdatesFormProps = {
-  buttonText: string;
-  heading: string;
-  subheading?: string;
+  type: 'contact' | 'registration' | 'newsletter';
   id: string;
+  buttonText: string;
+  target?: string;
+  redirect?: string;
 };
 
 export const ConferenceUpdatesForm = ({
-  value: { buttonText, heading, id },
-}: PortableTextComponentProps<ConferenceUpdatesFormProps>) => (
-  <GridWrapper>
-    <article className={styles.container}>
-      <PlaceholderImage width={330} height={459} className={styles.image} />
+  value: { type, id, buttonText, target, redirect },
+}: PortableTextComponentProps<ConferenceUpdatesFormProps>) => {
+  const router = useRouter();
 
-      <div className={styles.mainContents}>
-        <Heading type="h2">{heading}</Heading>
-        <form
-          action="https://sanity.us3.list-manage.com/subscribe/post"
-          method="POST"
-          className={styles.form}
-        >
-          <div className={styles.formContents}>
-            <input type="hidden" name="u" value={id} />
-            <input type="hidden" name="id" value="cca563332b" />
-            <input
-              id="mce-EMAIL"
-              name="EMAIL"
-              type="email"
-              placeholder="Your email address"
-              aria-label="Your email address"
-              required
-              className={styles.emailInput}
-            />
-            <button
-              type="submit"
-              className={styles.submitButton}
-              aria-label={buttonText}
-            >
-              -&gt;
-            </button>
-          </div>
-        </form>
-        <p className={styles.emailParagraph}>
-          We&#39;ll only send you updates about the conference
-        </p>
-      </div>
-    </article>
-  </GridWrapper>
-);
+  if (type !== 'newsletter') {
+    console.error(`Unrecognized ConferenceUpdatesForm type: '${type}'`);
+    return null;
+  }
+
+  return (
+    <GridWrapper>
+      <article className={styles.container}>
+        <PlaceholderImage width={330} height={459} className={styles.image} />
+
+        <div className={styles.mainContents}>
+          <form
+            action={target}
```

Comment:
Where does this `target` come from? It seems to be undefined, is that known?

Reply:
Nit: Should we output the form (or really, this section) at all if it doesn't lead anywhere?